plinth/modules/diaspora/__init__.py

```diff
@@ -0,0 +1,169 @@
+#
+# This file is part of Plinth.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import subprocess
+import os
+
+from django.utils.translation import ugettext_lazy as _
+
+from plinth.modules import names
+from plinth.utils import format_lazy
+from plinth import actions, action_utils, cfg, frontpage, service as service_module
+
+domain_name_file = "/etc/diaspora/domain_name"
+lazy_domain_name = None  # To avoid repeatedly reading from file
+
+
+def is_setup():
+    return os.path.exists(domain_name_file)
+
+
+def get_configured_domain_name():
+    if lazy_domain_name:
+        return lazy_domain_name
+
+    if not is_setup():
+        return ""
+
+    with open(domain_name_file) as dnf:
+        global lazy_domain_name
+        lazy_domain_name =  dnf.read().rstrip()
+        return lazy_domain_name
+
+
+version = 1
+
+title_en = 'Federated Social Network (diaspora*)'
+
+title = _(title_en)
+
+depends = ['apps']
+
+service = None
+
+managed_services = ['diaspora']
+
+managed_packages = ['diaspora']
+
+description = [
+    _('diaspora* is a decentralized social network where you can store and control your own data.'
+      ),
+    format_lazy(
+        'When enabled, the diaspora* pod will be available from '
+        '<a href="https://diaspora.{host}">diaspora.{host}</a> path on the web server.'.
```

Comment:
This produces a bad link if the domain name is not set. Perhaps a different message can be shown in that case.